build-aux/snap/snapcraft.yaml

```diff
@@ -420,14 +445,37 @@ parts:
       # snap should include the information about a particular openssl version
       # which was included in the build but the core22 FIPS variant snap has
       # been stripped off of the manifest.
-      if [ -f fips-build ]; then
+      if [ -f "${CRAFT_PROJECT_DIR}/fips-build-lp" ]; then
+        mkdir fips-libs-from-debs
+        mkdir fips-libs-from-debs-stage
+
+        # extract
+        for f in fips-debs/*.deb; do
+          dpkg -x "$f" fips-libs-from-debs
+        done
+
+        # and cherry pick the files we want
+        (
+          cd fips-libs-from-debs
+
+          # XXX note with -print0 only the last match is printed
+          find . \
+            -wholename './usr/lib/*-linux-gnu*/engines-3' \
+            -o -wholename  './usr/lib/*-linux-gnu*/ossl-modules-3' \
+            -o -wholename './usr/lib/*-linux-gnu*/libssl.so.3' \
+            -o -wholename './usr/lib/*-linux-gnu*/libcrypto.so.3' | \
+              xargs cp -av --parents --target ../fips-libs-from-debs-stage/
+        )
+        cp -av fips-libs-from-debs-stage/* "${CRAFT_PART_INSTALL}"
+      elif [ -f "${CRAFT_PROJECT_DIR}/fips-build" ]; then
```

Comment:
Can you add a comment that explains why we conceptually do this differently when on LP and when on non-LP?

Reply:
added